api-test/bdd/engine_test.go

```
@@ -0,0 +1,104 @@
+	out := os.Getenv("BDD_ENVELOPE_OUT")
+	if out == "" {
+		out = filepath.Join("..", "out", "bdd-envelope.json")
+	}
+	if err := writeEnvelope(out, rows); err != nil {
+		t.Fatalf("write envelope: %v", err)
+	}
+	t.Logf("bdd: wrote %d envelope row(s) to %s", len(rows), out)
+}
+
+func writeEnvelope(path string, rows []Envelope) error {
+	if err := os.MkdirAll(filepath.Dir(path), 0o755); err != nil {
```

golangci-lint (2.12.2)

[medium] G301: Expect directory permissions to be 0750 or less

(gosec)